libdrgn/program.c

```diff
@@ -207,12 +208,21 @@ drgn_program_check_initialized(struct drgn_program *prog)
 static struct drgn_error *has_kdump_signature(const char *path, int fd,
 					      bool *ret)
 {
-	char signature[KDUMP_SIG_LEN];
+	char signature[KDUMP_SIG_READ_LEN];
 	ssize_t r = pread_all(fd, signature, sizeof(signature), 0);
 	if (r < 0)
 		return drgn_error_create_os("pread", errno, path);
-	*ret = (r == sizeof(signature)
-		&& memcmp(signature, KDUMP_SIGNATURE, sizeof(signature)) == 0);
+	if (r != sizeof(signature)) {
+		*ret = false;
+		return NULL;
+	} else if (memcmp(signature, FLATTENED_SIGNATURE, FLATTENED_SIG_LEN) == 0) {
+		return drgn_error_create(DRGN_ERROR_INVALID_ARGUMENT,
+					 "The given file is in the makedumpfile flattened "
+					 "format, which Drgn does not support. Use "
+					 "'makedumpfile -R newfile <oldfile' to reassemble "
+					 "the file into a format Drgn can read.");
```

Comment:
Minor copy-editing for consistency: I don't capitalize drgn, and I don't usually capitalize or have periods in error messages. See for example: https://github.com/osandov/drgn/blob/579e68885a62819af05d4401262430715b68b102/libdrgn/program.c#L546-L552


```suggestion
		return drgn_error_create(DRGN_ERROR_INVALID_ARGUMENT,
					 "the given file is in the makedumpfile flattened "
					 "format, which drgn does not support; use "
					 "'makedumpfile -R newfile <oldfile' to reassemble "
					 "the file into a format drgn can read");
```